src/main/java/com/elastic/support/diagnostics/commands/RunKibanaQueries.java

```diff
@@ -0,0 +1,309 @@
+            for (int i = 0; i < actions.size(); i++) {
+                JsonNode config = actions.get(i).get("config");
+                // API webhook format can change, and maybe we have a webhook without config
+                if (!(config == null || config.isNull())) {
+                    // Not all the webhook need to have headers, so we need to be sure the data was set by the customer.
+                    JsonNode headers = actions.get(i).get("config").get("headers");
+                    if (!(headers == null || headers.isNull())) {
+                        Iterator<Map.Entry<String, JsonNode>> iter = actions.get(i).get("config").get("headers").fields();
+
+                        while (iter.hasNext()) {
+                            Map.Entry<String, JsonNode> entry = iter.next();
+                            if (!entry.getKey().equals("kbn-xsrf") && !entry.getKey().equals("Content-Type")) {
+                                iter.remove();
+                                headerRemoved = true;
+                            }
+                        }
+                    }
+                }
+            }
+            if (headerRemoved == true) {
+                String fileName = context.tempDir + SystemProperties.fileSeparator + "kibana_actions.json";
+                try (FileWriter fileWriter = new FileWriter(fileName)) {
+                    fileWriter.write(actions.toPrettyString());
+                    fileWriter.flush();
+                    fileWriter.close();
+                } catch (IOException e) {
+                  new RuntimeException("Error message", e).printStackTrace();
+                }
+            }
+        }
+    }
+
+
+    /**
+    * One of the requirements was test this new Kibana code.
+    * I splitted the "execute" function as in other examples (RunLogstashQueries) uses many static calls
+    * and create many new objects, make it difficult to mock and test.
+    * To be able to test the code without changing the global structure I splitted in smallest functions.
+    * Most of the new functions are public so I'm able to test it (Private if the test need to be done in a different class).
+    * When we will have more Unit test, the test for some functions here will be redundant and the public functions workaround will not needed anymore, 
+    * e.g. createJsonNodeFromFileName, this can be tested on the JsonYamlUtilsTest file (not existing in v8.1.2).
+    * 
+    *
+    * @param  DiagnosticContext context
+    * @return         JavaPlatform object
+    */
+    public void execute(DiagnosticContext context) {
+
+        try {
+            context.perPage         = 100;
+            RestClient client       = ResourceCache.getRestClient(Constants.restInputHost);
+            int totalRetries        = runBasicQueries(client, context);
+            allowedHeadersFilter(context);
+            execSystemCommands(context);
+
+        } catch (Throwable t) {
+            logger.error( "Kibana Query error:", t);
```

Comment:
It's unlikely that you want throwable. The only reasonably expected `Throwable` is `OutOfMemoryError`, which you will not be in a state to recover from given that `context` is manipulated as it goes due to the overall design of the existing diagnostic.

```suggestion
        } catch (Exception e) {
            logger.error("Kibana Query error:", e);
```